### PR TITLE
applications: serial_lte_modem: UART config/doc improvements

### PR DESCRIPTION
**Pull request by tomi-font**
See details in commit messages.

### CODE REVIEW

File: applications/serial_lte_modem/doc/CMUX_AT_commands.rst [hunk 1]
```diff
@@ -19,6 +19,18 @@ For example, it can be used to exchange AT data and have a :ref:`Point-to-Point
 CMUX is enabled in SLM by compiling it with the appropriate configuration files, depending on your use case.
 See the :ref:`slm_config_files` section for more information.
 
+.. slm_cmux_baud_rate_note_start
+
+.. note::
+
+   The maximum recommended baud rate is 460 800.
+   With higher baud rates (921 600, 1 000 000) it can happen that bytes come in faster than the chip is able to handle, and after a sufficient amount of them at once buffer space runs out.
```
Comment (wiba-nordic) [inline applications/serial_lte_modem/doc/CMUX_AT_commands.rst:27]:
```suggestion
   With higher baud rates (921 600, 1 000 000), it is possible for bytes to come in faster than the chip is able to handle, which can cause the buffer space to run out if it goes on for too long.
```

File: applications/serial_lte_modem/doc/CMUX_AT_commands.rst [hunk 2]
```diff
@@ -19,6 +19,18 @@ For example, it can be used to exchange AT data and have a :ref:`Point-to-Point
...
+   UART RX is not disabled in that case, which results in data loss and communication failures.
+
+   At a baud rate of 460 800, the maximum throughput is slightly below that of the nRF91 series modem.
```
Comment (wiba-nordic) [inline applications/serial_lte_modem/doc/CMUX_AT_commands.rst:30]:
```suggestion
   At a baud rate of 460 800, the maximum throughput is slightly below that of the nRF91 Series modem.
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

File: applications/serial_lte_modem/doc/CMUX_AT_commands.rst (hunk 2 above)
Comment (MarkusLassila) [reply applications/serial_lte_modem/doc/CMUX_AT_commands.rst:30]: ... modem with LTE-M1.

**Review by MarkusLassila — APPROVED** [review]
Looks good. I will run these over when I am setting CI up.

**Review by wiba-nordic — APPROVED** [review]

**Review by MarkusLassila — APPROVED** [review]

**Review by SeppoTakalo — APPROVED** [review]

**Pull request merged**